bloom/commands/generate.py

```diff
@@ -58,13 +58,21 @@ def load_generator_description(generator_name):
 
 def create_subparsers(parser, generator_cmds):
     metavar = '[' + ' | '.join(generator_cmds) + ']'
-    subparsers = parser.add_subparsers(
-        title='generate commands',
-        metavar=metavar,
-        description='Call `bloom-generate {0} -h` for help on a each generate command.'.format(metavar),
-        dest='generator_cmd',
-        required=True
-    )
+    if sys.version_info < (3, 7):
+        subparsers = parser.add_subparsers(
+            title='generate commands',
+            metavar=metavar,
+            description='Call `bloom-generate {0} -h` for help on a each generate command.'.format(metavar),
+            dest='generator_cmd'
+        )
+    else:
+        subparsers = parser.add_subparsers(
+            title='generate commands',
+            metavar=metavar,
+            description='Call `bloom-generate {0} -h` for help on a each generate command.'.format(metavar),
+            dest='generator_cmd',
+            required=True
+        )
```

Comment:
```suggestion
    subparsers_kwargs = {
      'title': 'generate commands',
      'metavar': metavar,
      'description': 'Call `bloom-generate {0} -h` for help on a each generate command.'.format(metavar),
      'dest': 'generator_cmd'
    }
    if sys.version_info >= (3, 7):
      subparsers_kwargs['required'] = True
    subparsers = parser.add_subparsers(**subparsers_kwargs)
```

What do you think of this as a way to avoid duplicating most of the arguments. I'm worried that with the current split it would be easy to update one of these calls and forget to do the other (even though they're currently adjacent).

Reply:
Thanks for the review @nuclearsandwich, I didn't think of that, and I am fine with either.
I think the original changes will likely end up with simpler code when time comes to remove the if statement, but as you said, someone may update one and forget about the other.

With the suggested change, it is less likely to have issues forgetting to synchronize changes in the two places, but when the if statement is removed in the future, it may leave the code unintuitive.

Reply:
> With the suggested change, it is less likely to have issues forgetting to synchronize changes in the two places, but when the if statement is removed in the future, it may leave the code unintuitive.

We do use this pattern across the infrastructure packages so I have confidence that later updates to remove the conditional would also restore the simpler inline kwargs.

Reply:
Sounds good - will commit the suggestion